V1.12.1 merge to master (#1198)
zowe/vscode-extension-for-zowe: packages/zowe-explorer/CHANGELOG.md

All notable changes to the "vscode-extension-for-zowe" extension will be documented in this file.

## `1.12.1`

- Fixed the issue that prevented edited profile base paths from being saved [#989](https://github.com/zowe/vscode-extension-for-zowe/issues/989). Thanks @katelynienaber
- Fixed the issue that prevented Zowe Explorer from storing empty values for optional profile fields, such as `user`, `password`, `timeout`, and `encoding`. This is done to be consistent with the way Zowe CLI stores profile information when creating and editing profiles [#1016](https://github.com/zowe/vscode-extension-for-zowe/issues/1016). Thanks @katelynienaber
- Fixed the issue that caused repeated credential prompting if a user refused to authenticate [#1147](https://github.com/zowe/vscode-extension-for-zowe/issues/1147). Thanks @katelynienaber

> - Fixed the issue that caused removed favorite profiles to be favorited again in subsequent IDE sessions [#1144](https://github.com/zowe/vscode-extension-for-zowe/issues/1144). Thanks @lauren-li

Reviewer: ```suggestion
- Fixed the issue that caused favorite profiles that were removed to persist between IDE sessions [#1144](https://github.com/zowe/vscode-extension-for-zowe/issues/1144). Thanks @lauren-li
```

Author: @IgorCATech This will be revised in v.13. Today, I am just going to add the changes from v.12.x branch.  But this is noted
